extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -89,30 +90,119 @@ public Signature boundSignature() {
     public Object execute(int idxInPartition,
                           WindowFrameState currentFrame,
                           List<? extends CollectExpression<Row, ?>> expressions,
-                          Input... args) {
-        final int offset;
-        if (args.length > 1) {
-            Object offsetValue = args[1].value();
-            if (offsetValue == null) {
-                return null;
+                          boolean ignoreNulls,
+                          Input[] args) {
+        calculateOffsetIfNull(args);
```

Comment:
I think this assumes that the offset is a static value - but I think it could be dynamic (e.g. based on another column) - in which case the value could change on each subsequent `execute` call.

Reply:
I think you are meaning for example, `lag(x,x)` where offset is actually a column. And I confirmed that it is supported by Postgres. But CrateDB currently only supports integers as an offset. 
> lag(argument any [, offset integer [, default any]])
I think this could be treated as a separate issue.